packages/interflex/src/translate.ts

```diff
@@ -0,0 +1,110 @@
+const PLACEHOLDER = /\{(\w+)\}/g;
+const FRAGMENT = /<(\w+)>(.*?)<\/\1>/g;
+const DEFAULT_DICTIONARY = {
+  localizedName: '',
+  currencySymbol: '',
+  thousandSeparator: '',
+  decimalSeparator: '',
+  decimalPlaces: 0,
+  dateFormat: '',
+  strings: {},
+  monthNames: [],
+  monthNamesShort: [],
+  daysOfWeek: [],
+  daysOfWeekShort: [],
+  startOfWeek: 0,
+};
+let dictionary: LocaleDictionary = {};
+let tLocale: Locale = 'en';
+
+export type Params = {
+  [key: string]:
+    | null
+    | undefined
+    | number
+    | string
+    | boolean
+    | ((s: string) => unknown);
+};
+
+/**
+ * (For testing) to inject data into the dictionary.
+ * @param data
+ */
+export function injectDictionaries(data: LocaleDictionary): void {
+  dictionary = { ...dictionary, ...data };
+}
```

Comment:
I guess this is just for testing, so it's not super important, but this seemed a bit confusing to me at first. Let's simplify it by taking two params:
```suggestion
export function injectLocale(locale: string, data: LocaleData): void {
  dictionary = { ...dictionary, [locale]: data };
}
```